r-package/grf/tests/testthat/test_boosted_regression_forest.R

```diff
@@ -0,0 +1,52 @@
+library(grf)
+
+set.seed(1)
+
+test_that("Boosted regression forest improves performance vs regular forest", {
+
+  n = 500; p = 6
+  X = matrix(runif(n * p), n, p)
+  mu = 2 * X[,1] * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y = mu + rnorm(n)
+  forest.regular <- regression_forest(X,Y)
+  forest.boost <- boosted_regression_forest(X,Y)
+
+  forest.Yhat <- predict(forest.regular)$predictions
+  boost.Yhat <- predict(forest.boost)$predictions
+
+  mse.forest <- mean((Y-forest.Yhat)^2)
+  mse.boost <- mean((Y- boost.Yhat)^2)
+  expect_true(mse.boost < mse.forest)
+})
+
+
+test_that("Boosted forest takes user specified number of steps",  {
+  n = 100; p = 6
+  X = matrix(runif(n * p), n, p)
+  mu = 2 * X[,1] * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y = mu + rnorm(n)
+  forest.boost <- boosted_regression_forest(X,Y,num.steps=2)
+  expect_equal(2,length(forest.boost$forests))
+})
+
+ test_that("Tolerance above 1 increases length of boosting", {
```

Comment:
Should we actually validate in `boosted_regression_forest` that `tolerance` is always less than 1? It doesn't seem to make a lot of sense to allow a number greater than 1 to be supplied.

Reply:
Agreed, I'll also remove this test.

Reply:
In the case you describe, could the user just use `boost.steps` to select the number of steps they want?

Reply:
You're right, added validation check to the input_utilities file